libcalico-go/lib/ipam/ipam_block.go

```diff
@@ -451,15 +505,21 @@ func (b allocationBlock) handleForIP(ip cnet.IP) (*string, error) {
 		log.Debugf("IP %s is not currently assigned in block", ip)
 		return nil, cerrors.ErrorResourceDoesNotExist{Identifier: ip.String(), Err: errors.New("IP is unassigned")}
 	}
-	return b.Attributes[*attrIndex].AttrPrimary, nil
+	if h := b.Attributes[*attrIndex].AttrPrimary; h != nil {
```

Comment:
We weren't hitting this before (and a couple of other new places that we call sanitize by handle) because we were never executing the `ReleaseIPs` code path (just `ReleaseByHandle`, which properly sanitized the input).